src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/Designer/ILaunchProfileEditorConnection.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug.Designer
+{
+    /// <summary>
+    /// Connection object provided to a launch profile editor client, via which a client
+    /// may add, remove, and update profiles.
+    /// </summary>
+    /// <remarks>
+    /// Must be disposed when the connection is no longer required.
+    /// </remarks>
+    internal interface ILaunchProfileEditorConnection : IAsyncDisposable
+    {
+        Task SetValuesAsync(Guid correlationId, ImmutableArray<PropertyValueChangeRequest> changes);
+
+        /// <summary>
+        /// Returns the set of available debug command names.
+        /// </summary>
+        Task<ImmutableArray<string>> GetDebugCommandNamesAsync();
```

Comment:
The assumption here is that once the debug command name is chosen, the user cannot change it. This is because the set of properties shown to the user depends on the debug command name.

Reply:
In the current implementation, the "command name" uniquely identifies who gets to convert the launch settings into actual debug settings, and the "owner" of the command gets to decide what properties to expose in the UI. This is a MEF-extensible system; we (.NET Project System) provide handlers for "Project" and "Executable", the Xamarin team provides a handler for "Xamarin", and the Web Tools team provides handlers for "IIS" and "IISExpress", for example. So we already have the possibility for a collision though in practice it is unlikely.

I'm not sure why `ILaunchProfile.CommandName` is nullable. It may be to allow us to handle the case where the command name is not present in the launchSettings.json file for whatever reason. Regardless, at this layer we can count on the command name being non-null--the implementation can always provide "Project" as a reasonable default. 

All that being said, I'll update the API to use `ProfileType` as you've laid it out here as it does make sense to provide help, a display name, and an icon.